### DISCUSSION TITLE
Difference between hold and wait

### DISCUSSION THREAD

**Discussion by chronvas** [holgerbrandl/kalasim#61] [Q&A] [answered]
What is the difference between hold and wait? Could they be named more succinctly, or any other word alternatives? Pause with parameters maybe?

**Comment by holgerbrandl** [comment, marked as answer]
See
https://www.kalasim.org/state/#process-interaction-with-wait
and
https://www.kalasim.org/component/#hold

Admittedly, in particular `hold` could be emphasized better in the docs, as it is the most important method in the entire API.

The names were simply kept when porting a python library salabmin to kotlin when creating kalasim.  `pause` does not sound much better to me compared to `hold` because all suspension functions in the `Component` API do _pause_ in some way or another. But if there's a sound argument, the naming could be changed to ease the usage of the library.
